ambari-server/src/main/java/org/apache/ambari/server/controller/internal/URLStreamProvider.java

```diff
@@ -323,7 +325,39 @@ protected HttpsURLConnection getSSLConnection(String spec) throws IOException, I
         .openConnection());
 
     connection.setSSLSocketFactory(sslSocketFactory);
- 
+
     return connection;
   }
+
+  /**
+   * A default implementation of {@link HttpURLConnectionProvider}, this class
+   * will use the {@link URLStreamProvider} in order to provide an
+   * {@link HttpURLConnection} which is able to use Ambari's cookie store,
+   * truststore, and timeout values.
+   */
+  public static final class AmbariHttpUrlConnectionProvider implements HttpURLConnectionProvider {
```

Comment:
Why is this not in its own class file?

Reply:
There's really nothing to this class. I didn't want to create more bloat in the Utils package, and since `AmbariHttpUrlConnectionProvider` is backed by a `URLStreamProvider`, I figured here's a good place to put it. 

Would you prefer I moved it out?

Reply:
I have no preference... just making sure this was on purpose.